erpnext/assets/doctype/asset/asset.py

```diff
@@ -734,6 +739,22 @@ def make_gl_entries(self):
 			make_gl_entries(gl_entries)
 			self.db_set("booked_fixed_asset", 1)
 
+	def check_asset_capitalization_gl_entries(self):
+		if self.is_composite_asset:
+			asset_capitalization, target_fixed_asset_account = frappe.db.get_value(
+				"Asset Capitalization",
+				{"target_asset": self.name, "docstatus": 1},
+				["name", "target_fixed_asset_account"],
+			)
+
+			# Fetch GL entries for the retrieved Asset Capitalization
+			gl_entries_test = get_gl_entries("Asset Capitalization", asset_capitalization)
+
+			# Return if GL entry's account matches the target fixed asset account
+			if gl_entries_test and gl_entries_test[0].account == target_fixed_asset_account:
```

Comment:
There are chances of debiting multiple accounts from asset capitalization. Hence, don't compare zero-th element to target_fixed_asset_account. Instead, `get_gl_entries` should return gle for only target_fixed_asset_account.